### ISSUE TITLE
Implement PUT projects/:projectID/files/:fileID, /files/:fileID

### ISSUE THREAD

**Issue by gideonthomas** [mozilla/publish.webmaker.org#37]
Update the contents of a file by sending the following message body:
```javascript
{ 
   id: ...,
   name: ...,
   path: ...,
   buffer: ...
}
```

**Comment by gideonthomas** [comment]
We might have an issue implementing this and #36 on the client-side because (normally), Filer can't differentiate between a file-create and a file-update; so, we might not have enough info to decide whether to hit the POST route or the PUT route.
Since we have auto-saving enabled on the client, maybe we can do something like do an `fs.exists` check to determine whether to send a POST or PUT request before the `writeFile`.
Another thing that came up was whether we want to persist empty files? If so, how do we do it (store empty buffers in the db / populate the buffer with something to indicate an empty file / have a flag in the db that indicates whether that file is empty or not)?

cc @humphd, @sedge, @ashleygwilliams

**Comment by humphd** [comment]
Stat-before-write is the way Brackets works now, so it's fine if you want to do this.

**Comment by ashleygwilliams** [comment]
there will be 2 routes to create new files:

```
PUT projects/:project_id/files/:file_id
POST files/:file_id
should receive:
```

```
{
   path: String,
   project_id: Integer,
   buffer: Blob
}
```

**Comment by ashleygwilliams** [comment]
this is solved by #44
